### ISSUE TITLE
Initial Questions & Project Organization

### ISSUE THREAD

**Issue by anthonyLuttrell** [warofants/wwa#5]
INITIAL QUESTIONS:
--------------------------
What kind of game is this?
What base classes do we need to write before we can give out tasks to newbs?
What kind of development standards should we stick to?
What expectations do you have for a contributor?
What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
What is necessary for a text based game to be considered entertaining?
Will there be a story to this game?
What features will be included? hp display, sound fx, score, dialog?
What is winning? What is losing?

PROJECT ORGANIZATION:
---------------------------------
Set up roles such as Contributor, Reviewer, and Task Organizer
Set up a shared list of tasks that only task organizers can edit on the repository
-this list of tasks will be ordered task# and task category
-each branch submission should be labeled by the above identifiers and short description
-this will require a decent# of initial tasks in order to work
-each submission should be reviewed by a reviewer and update the task list with the user name and time the task was completed
Set up a suggestions channel to determine some tasks that we may miss higher up

**Comment by anthonyLuttrell** [comment]
From @Johnny

What kind of game is this?
It sounds to me that this would could be a tile based ant simulator, similar to Civilization but for ant colonies. What do you all think?

What base classes do we need to write before we can give out tasks to newbs?
The basic game loop should probably be written, and we should have some basic interfaces that allow for the MVP (Minimum Viable Product) to function.

These would include things like IMove, IGather, IFight, IFly, ISwim -- etc. Then we can implement an Abstract Base Class for Entities which represent any entity in the game, and inherited from that we'll have a Abstract Base Class for ants which all ants will inherit. Then we can look at what else we need from there.

What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
I'll need more input from others here, however I do recommend for command line input that we use a "Command" based pattern for UserInput. https://java-design-patterns.com/patterns/command/ 

What is necessary for a text based game to be considered entertaining?
It has to be fun! Haha, I mean look how many people play Dwarf Fortress.

What is winning? What is losing?
Since this is a war game, clearly one game mode would have to involve the destruction of all other colonies. Other victory modes could be included.

**Comment by Pacutacatete100** [comment]
What kind of game is this?
I agree with what @RedLoyalDog said, itd be cool to make it a type civilization for ants

What base classes do we need to write before we can give out tasks to newbs?
game loop, interfaces or abstract classes for ant types (army ants and all that), noobs can extend on those

What expectations do you have for a contributor?
make at least one class or something, something meaningful to the game, maybe an avid debugger in the program if they dont want to do that, (i.e. something important and meaningful)

What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
agree as to what @RedLoyalDog says

What is necessary for a text based game to be considered entertaining?
Give the user enough choices and gameplay so it doesnt feel like theya re just watching a story go by

Will there be a story to this game?
I dont think a story at this stage could be too necessary, maybe just like a casual game, sot he user doesnt have to worry about a story and all that, but it could be cool if you could start your own colony and evolve that and expand that

What features will be included? hp display, sound fx, score, dialog?
for now, text base will be fine, in the future we could make a 2d graphics game with some music and a few sound effects, graphics would be cool of they are inspired or based off the kursgesagt video this whole thing was inspired by

What is winning? What is losing?
since imo it should a casual game, there is no win goal, but losing would be losing your entire colony or close to it

**Comment by NishantAcharya** [comment]
Well here is my take on the questions:
What kind of game is this? - Text based Civ style game, I believe we all agreed on that
What base classes do we need to write before we can give out tasks to newbs? - lets make the abstract classes for each species, then hand over the subspecies development with instructions
What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/ - We are still deciding that I believe
What is necessary for a text based game to be considered entertaining? - well it's a text based strategy game so, its like chess in a way, we can add a bit of background story if we want.
Will there be a story to this game? - nope, we can add one but the story will be outlandish
What is winning? What is losing? - This is debatable but winning an ant colony in nature means either remove the queen or kill everyone in the colony

**Comment by LoveTheAardvarks** [comment]
INITIAL QUESTIONS:
What kind of game is this?
A tile based text game based on ant colonies that combines resource management with turn based combat and exploration.

What base classes do we need to write before we can give out tasks to newbs?
I'm the newb. But I'm great at working out what I need after the task is given to me.

What expectations do you have for a contributor?
I don't necessarily think everyone needs to contribute code. Ideas work, concept art, etc. But the people who do code should comment their code and the end product should be in the same format(spacing vs tabs, var names etc.).

Will there be a story to this game?
I'm not sure it's necessary. I'm personally seeing it as a short match kind of game.

What features will be included? hp display, sound fx, score, dialog?
Number of workers, number of territory blocks, food available/ant in colony, attack damage/defense, action dialogue. I think sound effects can come when a gui is considered.

What is winning? What is losing?
winning can be taking out the enemy colonies/s or controlling more than a certain % of the board. Maybe player chooses win conditions.

**Comment by KrakenToad** [comment]
What kind of game is this?
Civ style ant warfare game
What base classes do we need to write before we can give out tasks to newbs?
abstract classes for ants, whatever @royalreddog said

What expectations do you have for a contributor?

What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
What is necessary for a text based game to be considered entertaining?
people actually want to play it and it doesnt have too many obvious bugs
Will there be a story to this game?
YES!!! Game of Thrones with ANTS...see some ant queen boobies :)
What features will be included? hp display, sound fx, score, dialog?
hp, awsome 8-bit sound and music, flashing lights that give young kids seizures, and queen ant boobies, 
What is winning? What is losing?
Your colony get destoryed you lose, you succesfully raid the other colonies or picnic baskets, you win

**Comment by TrainingAfternoon** [comment]
_What kind of game is this?_ By popular agreement, CIV-like with text graphics

_What base classes do we need to write before we can give out tasks to newbs?_ Abstract Entity class as a base, which can then branch out into an Abstract for Ant / Wasp / Termite / etc. Assuming we're going with a Friend / Foe / Food / NPC system, probably an enum for those types.

_What kind of development standards should we stick to?_ I'll leave this to others who know more about development standards than I do.

_What expectations do you have for a contributor?_ Functioning code that follows whatever standards are set. It doesn't have to be perfect, but an honest attempt should be made.

_What is the scope of this project?_ look at this https://www.makeuseof.com/tag/browser-text-based-games/ A turn-based, CIV-like, sandbox where the goal is survival.

_What is necessary for a text based game to be considered entertaining?_ Entertaining gameplay, engaging story, or creative graphic design. Preferably all three qualities.

_Will there be a story to this game?_ I hope so. The game doesn't necessarily need a campaign mode, but little tidbits of world building sprinkled about in discoverable artifacts would be nice.

_What features will be included?_ hp display, sound fx, score, dialog? I think we should have a combat system, a building/resource-gathering system, and some sort of social system. The particulars would have to be worked out after a general idea on how these systems should function is decided on.

What is winning? What is losing? Winning is survival and losing is death. How the player survives can be up them. Amass strength to crush enemies, use diplomacy to form defense pacts, build an impenetrable fortress, etc.

**Comment by ricjohn-acosta** [comment]
**What kind of game is this?**
- I can see this game as a survival, sandbox, maybe even multiplayer, top down game (something like Don't Starve Together). 

**What base classes do we need to write before we can give out tasks to newbs?**
- Definitely an Abstract Entity class that Abstract Enemy,  Abstract Ally, Abstract NPC will extend to. 

**What kind of development standards should we stick to?**
- At the very least, it would be great to have a certain style guide we will all refer to. "soft" deadlines would be good to have too just to keep that element of time pressure in.

**What expectations do you have for a contributor?**
- I expect contributors to be able to pitch ideas as we go and contribute somewhat readable code.

**What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/**
- Not really sure what this means but having a console-based game first and then a GUI would be ideal.

**What is necessary for a text based game to be considered entertaining?**
- This project will have a GUI so it's not entirely going to be text-based but the console-based version of this game should have the functionalities that the final product with a GUI would have.

**Will there be a story to this game?**
- a non-serious story filled with subtle dark-dry humor like fallout would be fun :) 

**What features will be included? hp display, sound fx, score, dialog?**
1. Randomly generated entities
2. Movement
3. Combat - Health, Stamina?
4. Building

What is winning? What is losing?
- Death = lost but can respawn, Winning = no winning, there would be bosses so somehow it's gonna be like minecraft.

**Comment by lemikistu** [comment]
Trello board link: https://trello.com/invite/b/xDj6KfHq/372d3221c238af4dbc52cf65659a3c6e/world-war-of-ants

**Comment by XanderRoy** [comment]
What kind of game is this?
Civ 5 meets Oregon trail meets overlord
Control your colony as you scavenge for survival.  Collect more biomass to get upgrades.  Evolve a unique colony and try to survive as long as possible. 

What base classes do we need to write before we can give out tasks to newbs?
I imagine none? If a task is too hard for a new should be easy to get help.




What is necessary for a text based game to be considered entertaining?
Multiple paths!!

Will there be a story to this game?
Survive the collapse of your ecosystem. 
 
What features will be included? hp display, sound fx, score, dialog?
Leaderboards for longest lived species
What is winning? What is losing?
 You  could win if your ant colony evolves infinite energy, or escapes into the internet. A diplomatic victory should be possible.

**Comment by DavidHanover** [comment]
What kind of game is this?
Simple 1v1 'battle' rpg.  Colony vs colony. Turn based with experience and leveling per colony.  Experience can either add more ants (thus functioning like a level system) or give new moves to colonies.  

What base classes do we need to write before we can give out tasks to newbs?
Ant, Colony, BattleLoop/Game

What kind of development standards should we stick to?
up to you.  I'll follow whatever, unopinionated

What expectations do you have for a contributor?
none whatsoever

What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
sure, seems about right

What is necessary for a text based game to be considered entertaining?
if it works or not

Will there be a story to this game?
no, just a battle system where you can challenge other colonies.  maybe world colonies are stored on a server somewhere & you can just challenge random others within your level/ant count range?  or maybe challenges depend on win/loss ratio?  or win streaks?  A pvp mode would also be cool, but would require some networking and server coders.  Also, a 1 vs random game design would require some kind of AI.  maybe a bit out of scope of this?  dunno

What features will be included? hp display, sound fx, score, dialog?
Colony HP, colony attack/defense moves, Individual Ant level, class, HP, count, attack stat, defense stat, special attack stat, special defense stat, maybe ant types.  think Pokemon-light

What is winning? What is losing?
drain the enemy colony's HP to 0, while keeping your own colony's HP above 0

**Comment by Raveesh** [comment]
What kind of game is this?
A cli driven game with a very minimalistic ui.
And an option to turn off ui completely. 
It's a survival of the fittest colony. 
With options to play against the computer or other players.

Hoping that each player will first be guided through which colony of ant he prefers each one listed with strengths, traits and weakness and threats . 

What base classes do we need to write before we can give out tasks to newbs?

Interfaces related to moves that are allowed 
Interfaces for species , colonies 
Default abstract implementation  for players who are unable to make a choice

What kind of development standards should we stick to?
Clean code guidelines including variable names , solid principles, KISS and DRY

What expectations do you have for a contributor?
Every checkin should have Unit tests or preview of what will be available . Code should have tests and a report indicating 100% pass rate

What is the scope of this project? look at this https://www.makeuseof.com/tag/browser-text-based-games/
Multiplayer or play against the computer.
Should be adventure / exploratory where computer can simulate attacks at random points in time .. not in multiplayer scenario.

Points to be given based on factors like who captures and kills the queen . Who pushes through the first line of defense and in how much time etc

What is necessary for a text based game to be considered entertaining?

Evolution of new unexpected scenarios can keep players hooked to the game.
Nuggets of life and hidden gems will make players explore every crevice of the game

Will there be a story to this game?
I am thinking more from a strategy and holding for perspective if the game is in adventure mode.
If its exploratory mode then choices become unlimited .story may not be necessary 

What features will be included? hp display, sound fx, score, dialog?
All of the above .. with options to turn off all of the above.:)
What is winning? What is losing?
Everyone looses at some point in time and that's where the fun is :)

**Comment by blackk100** [comment]
**What kind of game is this?**
Text-based sandbox-ish ant colony management and war simulator (oof that's a lot of words).
However, at what level of abstraction of control?
- _Civilisation/StarCraft/Age of Emipres-like_: You control everything. Would be easier to make in a GUI based interface I guess, not sure if that would be the case in a text-based interface.
- _Dwarf Fortress/Rimworld-like_: You assign tasks to be done, the inhabitants will do it if they feel like it. It may be difficult to implement depending on the number of subsystems we'll have.
- _[Warsim](https://store.steampowered.com/app/659540/Warsim_The_Realm_of_Aslona/)-like_: A mix of both above; you control the head of the colony and tell your subjects what to do. I personally like this method.

**What base classes do we need to write before we can give out tasks to newbs?**
In my opinion, nothing. If the intention is for letting others learn while working on a real project, everything should be done by those who are going to do the work, with assistance/advice wherever required. Tasks could be segregated based on who volunteers, their experience, etc. (For example: Simpler things like species, structures, etc. could be done by a group of newbies while more complex things like world generation, game loop, user interfaces, etc. could be done by a group of more experienced coders).
Only the initial theoretical design should be pre-defined, but no actual code before the "work" starts.

**What kind of development standards should we stick to?**
A style guide at a minimum, and since we're all learning, well-documented code. I second @ricjohn-acosta 's suggestion on having soft deadlines as well.

**What expectations do you have for a contributor?**
Following the project's guidelines and finishing tasks within the deadline set.

**What is the scope of this project?**
A full-fledged text-based game initially. Could be expanded to a TUI (text user interface), then a GUI, and finally 2D graphics. Also create "replicas" in languages other than Java (Python, C/C++, C#, Go, JS, etc.), preferably simultaneously.

**What is necessary for a text-based game to be considered entertaining?**
Things that make me interested/disinterested in a text-based game include whether or not it keeps me engaged, the learning curve in playing it, the amount of time I will have to invest into playing it satisfactorily (i.e., whether it is the easy to learn, hard to master-type; just easy all-around; just hard all-around; etc.), and the interface (is it like a terminal, command-based input; or does it use a TUI; or something else. Is the interface used intuitive enough for the game and does it enhance the game's appeal (like how [Duskers](https://store.steampowered.com/app/254320/Duskers/) has a command based input but a 2D GUI output)).

**Will there be a story to this game?**
An initial/barebones lore could be present, but it shouldn't be a priority initially. After the game starts becoming playable, the story could be developed simultaneously.
A counterpoint though could be that having good lore beforehand could help in deciding what new things could be added/old things can be overhauled.

**What features will be included? hp display, sound fx, score, dialogue?**
- HP or body part-based damage?
- Sound FX would be nice.
- Randomized worlds (seed strings like Minecraft?)
- AI-controlled colonies developing alongside the players (no hacks though. The AI should be considered equivalent to an actual player by the Game Engine).
- Turn-based or real-time?

**What is winning? What is losing?**
[Losing is fun!](http://dwarffortresswiki.org/index.php/DF2014:Losing)
There could be various game modes, with each mode having a separate win condition, alongside a pure-sandbox mode where you just survive for as long as possible (like Dwarf Fortress' Fortress Mode). An idea to consider would be Achievements; it would give players a sense of direction, but can "distract" them from actual sandbox gameplay (at least that's what the Kerbal Space Program devs think).

**Comment by RedRoyalDog** [comment]
Going to close this, as it has been open for almost a month.

Please checkout the discord for the several documents that discuss this.
